xbmc/cores/DllLoader/exports/emu_msvcrt.cpp

```diff
@@ -1881,19 +1881,14 @@ extern "C"
 
           if (free_position != NULL)
           {
-            // free position, copy value
             size = strlen(var) + strlen(value) + 2;
-            *free_position = (char*)malloc(size); // for '=' and 0 termination
-            if ((*free_position))
+            *free_position = static_cast<char*>(malloc(size));
+            if (*free_position)
             {
-              strncpy(*free_position, var, size);
-              (*free_position)[size - 1] = '\0';
-              strncat(*free_position, "=", size - strlen(*free_position));
-              strncat(*free_position, value, size - strlen(*free_position));
+              snprintf(*free_position, size, "%s=%s", var, value);
```

Comment:
Replace `strncpy` to `strncpy_s` and `strncat` to `strncat_s` is also an option:

```c++
              strncpy_s(*free_position, size, var, size);
              (*free_position)[size - 1] = '\0';
              strncat_s(*free_position, size, "=", size - strlen(*free_position));
              strncat_s(*free_position, size, value, size - strlen(*free_position));
```

Reply:
I think the `snprintf` helps a lot with readability here.